core/src/main/java/org/hisp/dhis/android/persistence/common/querybuilders/ReadOnlySQLStatementBuilderImpl.kt

```diff
@@ -110,6 +111,16 @@ internal open class ReadOnlySQLStatementBuilderImpl(
         return RoomRawQuery("DELETE FROM $tableName WHERE $whereClause;")
     }
 
+    override fun updateWhere(updates: ArrayMap<String, Any>, whereClause: String): RoomRawQuery {
+        val setClause = updates.entries.joinToString(", ") { "${it.key} = ${formatValue(it.value)}" }
```

Comment:
I think you would need to wrap the value with single quotes because it is a string literal. In the case of numbers, it wouldn't be required.

The class ContentValues handles this formatting automatically.